cli/cmd/proxy/read/command_test.go

```diff
@@ -146,6 +140,97 @@ func TestReadCommandOutput(t *testing.T) {
+// field filter (e.g. -fqdn default) and a table filter (e.g. -secrets) where
+// the former does not affect the output of the latter.
+func TestFilterWarnings(t *testing.T) {
+	podName := "fakePod"
+	cases := map[string]struct {
+		input      []string
+		shouldWarn bool
+	}{
+		"fully qualified domain name doesn't apply to listeners": {
+			input:      []string{"-fqdn", "default", "-listeners"},
+			shouldWarn: true,
+		},
+		"fully qualified domain name doesn't apply to routes": {
+			input:      []string{"-fqdn", "default", "-routes"},
+			shouldWarn: true,
+		},
+		"fully qualified domain name doesn't apply to endpoints": {
+			input:      []string{"-fqdn", "default", "-endpoints"},
+			shouldWarn: true,
+		},
+		"fully qualified domain name doesn't apply to secrets": {
+			input:      []string{"-fqdn", "default", "-secrets"},
+			shouldWarn: true,
+		},
+		"fully qualified domain name doesn't apply to endpoints or listeners": {
+			input:      []string{"-fqdn", "default", "-endpoints", "-listeners"},
+			shouldWarn: true,
+		},
+		"fully qualified domain name doesn't apply to listeners, routes, endpoints, or secrets": {
+			input:      []string{"-fqdn", "default", "-listeners", "-routes", "-endpoints", "-secrets"},
+			shouldWarn: true,
+		},
+		"port doesn't apply to routes": {
+			input:      []string{"-port", "8080", "-routes"},
+			shouldWarn: true,
+		},
+		"port doesn't apply to secrets": {
+			input:      []string{"-port", "8080", "-secrets"},
+			shouldWarn: true,
+		},
+		"port doesn't apply to secrets or routes": {
+			input:      []string{"-port", "8080", "-secrets", "-routes"},
+			shouldWarn: true,
+		},
+		"address does not apply to routes": {
+			input:      []string{"-address", "127.0.0.1", "-routes"},
+			shouldWarn: true,
+		},
+		"address does not apply to secrets": {
+			input:      []string{"-address", "127.0.0.1", "-secrets"},
+			shouldWarn: true,
+		},
+		"multiple warnings": {
+			input:      []string{"-address", "127.0.0.1", "-port", "8080", "-secrets"},
+			shouldWarn: true,
+		},
+		"no warning produced (happy case)": {
+			input:      []string{"-fqdn", "default", "-clusters"},
+			shouldWarn: false,
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			fakePod := v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      podName,
+					Namespace: "default",
+				},
+			}
+
+			buf := new(bytes.Buffer)
+			c := setupCommand(buf)
+			c.kubernetes = fake.NewSimpleClientset(&v1.PodList{Items: []v1.Pod{fakePod}})
+			c.fetchConfig = func(context.Context, common.PortForwarder) (*EnvoyConfig, error) {
+				return testEnvoyConfig, nil
+			}
+
+			exitCode := c.Run(append([]string{podName}, tc.input...))
+			require.Equal(t, 0, exitCode) // This shouldn't error out, just warn the user.
+
+			if tc.shouldWarn {
+				require.Regexp(t, "The filter.*does not apply to the tables displayed.", buf.String())
+			} else {
```

Comment:
how do we check that multiple warnings are displayed? Also, could we check for the specific warning string that's applicable to each case?

Reply:
I improved it to check for a list of warnings so we can confirm multiple warnings are produced. :)